jans-linux-setup/jans_setup/setup_app/utils/base.py

```
@@ -381,20 +382,42 @@ def mylog(logs):
             urllib.request.urlretrieve(url, dst)
             mylog("Download size: {} bytes".format(os.path.getsize(dst)))
             time.sleep(0.1)
+            download_ok = True
         except:
-             mylog("Error downloading {}. Download will be re-tried once more".format(url))
-             download_tries += 1
-             time.sleep(1)
+            retry_sec = random.randint(1,4)
+            mylog(f"Error downloading {url}. Download will be re-tried once more in {retry_sec} seconds.")
+            download_tries += 1
+            time.sleep(retry_sec)
```

Ruff (0.15.21)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)